source/server/filter_chain_manager_impl.h

```diff
@@ -101,8 +102,18 @@ class FilterChainImpl : public Network::DrainableFilterChain {
   std::chrono::milliseconds transportSocketConnectTimeout() const override {
     return transport_socket_connect_timeout_;
   }
-  const std::vector<Network::FilterFactoryCb>& networkFilterFactories() const override {
-    return filters_factory_;
+  std::vector<Network::FilterFactoryCb> networkFilterFactories() const override {
+    std::vector<Network::FilterFactoryCb> filters;
+    filters.reserve(filters_factory_.size());
+    for (const auto& provider : filters_factory_) {
+      auto filter_factory = provider->config();
+      if (filter_factory.has_value()) {
+        filters.push_back(filter_factory.value());
+      } else {
+        return {};
+      }
+    }
+    return filters;
```

Comment:
networkFilterFactories() is a breaking change.
Previously FilterFactoryCb can be lambda that captures states. Those states are valid within the life of the network filter chain.
With this PR, the `FilterFactoryCb`s are fired and forgot. Some network filters may survive but probably not all. See also #14711 that utilizes the above assertion.

Reply:
I am not sure I understand. Is there a difference with using static filter factory provider that wraps the callback vs using the callback directly?

Reply:
Should the functors really own the config state though? The common style in envoy is to pass functors by value. I can change it to pass by reference but that sounds odd to me.